### ISSUE TITLE
Expression evaluation API with timeout behaves incorrectly

### ISSUE THREAD

**Issue by suhreibm** [IBM/JSONata4Java#245]
When calling `Expressions.evaluate(JsonNode rootContext, long timeoutMS, int maxDepth)` and more time than `timeoutMS` has passed when you call it again on the same object, an `EvaluateException` with the message `Expression evaluation timeout: Check for infinite loop` is thrown. Seems like the start time to measure the timeout is not the start of each `evaluate` call but only the first one.

Here is a simple testcase (Note that the first call uses `evaluate(JsonNode)` without the timeout).

```
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import com.api.jsonata4java.expressions.Expressions;
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;

public class JsonataTimeoutTest {

    @Test
    void testTimeouts() throws Exception {
        long timeout = 100;
        int maxDepth = 100;
        JsonNode jn = new ObjectMapper().readTree("{ \"a\": \"stuff\" }");

        Expressions expr = Expressions.parse("{ \"b\": a }");
        JsonNode result1 = expr.evaluate(jn);
        Assertions.assertEquals("stuff", result1.get("b").asText());

        // sleep some time longer than the timeout
        Thread.sleep(timeout * 2);

        JsonNode result2 = expr.evaluate(jn, timeout, maxDepth); // this throws the EvaluateException
        Assertions.assertEquals("stuff", result2.get("b").asText());
    }
    
}
```

I checked the code a little and it seems that `ExpressionsVisitor.startTime` is only set when the object is created. Rather, it should be set when an evaluation is started.

**Comment by wnm3** [comment]
Thanks for the example. I've added it to the test suite, and fixed the problem. I've created and released version 2.2.6 which should be in maven in a couple of hours. Please close if you agree the problem has been fixed.

**Comment by suhreibm** [comment]
Ran the testcase with 2.2.6 and it worked. Closing issue.
Thanks @wnm3 For the very quick turnaround!
